QuasarHelios2015/src/org/team1540/quasarhelios/Clamp.java

```diff
@@ -48,25 +53,22 @@ public static void setup() {
 
         heightReadout = FloatMixing.normalizeFloat(encoder, min, max);
 
-        PIDControl pid = new PIDControl(heightReadout, height, p, i, d);
+        PIDControl pid = new PIDControl(heightReadout, heightOrSpeed, p, i, d);
```

Comment:
Potential issue: what if the integral term gets really large?

* Maybe you should reset the accumulator whenever we enter height mode. (`PIDControl.integralTotal`)
* Maybe set bounds on the integral term. (`setIntegralBounds`)